### PR TITLE
clarify UDF overloading in documentation

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/2.3.0.rst [hunk 1]
```diff
@@ -59,6 +59,12 @@ Breaking Changes
   ``PRIMARY_KEY`` has been changed to ``PRIMARY KEY``. Also PRIMARY KEY
   constraint is not returned when not explicitly defined.
 
+- Scalar functions are resolved more strictly based on the argument types,
+  meaning that built-in functions with the same name as user defined functions
+  will always "hide" the latter, even if the UDF has a different set of
+  arguments. Using the same name as a built-in function for a user defined
+  function is cosidered bad practice!
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/2.3.0.rst]: typo in `considered`

File: blackbox/docs/general/user-defined-functions.rst [hunk 2]
```diff
@@ -28,24 +28,24 @@ full syntax description.
...
 
 ::
 
-    cr> SELECT subtract(3, 1);
-    +----------------+
-    | subtract(3, 1) |
-    +----------------+
-    |              2 |
-    +----------------+
+    cr> SELECT my_subtract_function(3, 1);
```
Comment (mfussenegger) [inline blackbox/docs/general/user-defined-functions.rst]: I'd also add the schema to make the built-in lookup explicit to follow best-practice. If users don't do that they're always risking that new built-ins that we add end up clashing with their functions and break their stuff.

File: blackbox/docs/appendices/release-notes/2.3.0.rst [hunk 3]
```diff
@@ -59,6 +59,12 @@ Breaking Changes
   ``PRIMARY_KEY`` has been changed to ``PRIMARY KEY``. Also PRIMARY KEY
   constraint is not returned when not explicitly defined.
 
+- Scalar functions are resolved more strictly based on the argument types,
+  meaning that built-in functions with the same name as user defined functions
+  will always "hide" the latter, even if the UDF has a different set of
+  arguments. Using the same name as a built-in function for a user defined
+  function is considered bad practice!
```
Comment (nslater) [inline blackbox/docs/appendices/release-notes/2.3.0.rst]: replace "!" for "."

File: blackbox/docs/appendices/release-notes/2.3.0.rst [hunk 4]
```diff
@@ -59,6 +59,12 @@ Breaking Changes
   ``PRIMARY_KEY`` has been changed to ``PRIMARY KEY``. Also PRIMARY KEY
   constraint is not returned when not explicitly defined.
 
+- Scalar functions are resolved more strictly based on the argument types,
+  meaning that built-in functions with the same name as user defined functions
```
Comment (nslater) [inline blackbox/docs/appendices/release-notes/2.3.0.rst]: "... argument types. This means that ..."

**Comment by chaudum** [comment]
@nslater could you have another look please

**Pull request merged**